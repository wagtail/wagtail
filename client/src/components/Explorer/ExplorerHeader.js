import React from 'react';
import { ADMIN_URLS, STRINGS } from '../../config/wagtailConfig';

import Button from '../../components/Button/Button';
import Icon from '../../components/Icon/Icon';

/**
 * The bar at the top of the explorer, displaying the current level
 * and allowing access back to the parent level.
 */
const ExplorerHeader = ({ page, depth, onClick }) => {
  const isRoot = depth === 1;

  return (
    <Button
      href={page.id ? `${ADMIN_URLS.PAGES}${page.id}/` : ADMIN_URLS.PAGES}
      className="c-explorer__header"
      onClick={onClick}
    >
      <div className="c-explorer__header__inner">
        <Icon name={isRoot ? 'home' : 'arrow-left'} />
        <span>{page.title || STRINGS.PAGES}</span>
      </div>
    </Button>
  );
};

ExplorerHeader.propTypes = {
  page: React.PropTypes.shape({
    id: React.PropTypes.oneOfType([React.PropTypes.string, React.PropTypes.number]),
    title: React.PropTypes.string,
  }).isRequired,
  depth: React.PropTypes.number.isRequired,
  onClick: React.PropTypes.func.isRequired,
};

export default ExplorerHeader;
