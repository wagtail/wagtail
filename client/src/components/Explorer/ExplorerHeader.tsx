import React from 'react';
import { ADMIN_URLS } from '../../config/wagtailConfig';

import Button from '../../components/Button/Button';
import Icon from '../../components/Icon/Icon';
import { PageState } from './reducers/nodes';

interface SelectLocaleProps {
  locale?: string;
  translations: Map<string, number>;
  gotoPage(id: number, transition: number): void;
}

const SelectLocale: React.FunctionComponent<SelectLocaleProps> = ({
  locale,
  translations,
  gotoPage,
}) => {
  /* eslint-disable camelcase */
  const options = wagtailConfig.LOCALES.filter(
    ({ code }) => code === locale || translations.get(code),
  ).map(({ code, display_name }) => (
    <option key={code} value={code}>
      {display_name}
    </option>
  ));
  /* eslint-enable camelcase */

  const onChange = (e) => {
    e.preventDefault();
    const translation = translations.get(e.target.value);
    if (translation) {
      gotoPage(translation, 0);
    }
  };

  return (
    <div className="c-explorer__header__select">
      <select value={locale} onChange={onChange} disabled={options.length < 2}>
        {options}
      </select>
      <Icon name="arrow-down" className="c-explorer__header__select-icon" />
    </div>
  );
};

interface ExplorerHeaderProps {
  page: PageState;
  depth: number;
  onClick(e: any): void;
  gotoPage(id: number, transition: number): void;
}

/**
 * The bar at the top of the explorer, displaying the current level
 * and allowing access back to the parent level.
 */
const ExplorerHeader: React.FunctionComponent<ExplorerHeaderProps> = ({
  page,
  depth,
  onClick,
  gotoPage,
}) => {
  const isRoot = depth === 0;
  const isSiteRoot = page.id === 0;

  return (
    <div className="c-explorer__header">
      <Button
        href={!isSiteRoot ? `${ADMIN_URLS.PAGES}${page.id}/` : ADMIN_URLS.PAGES}
        className="c-explorer__header__title"
        onClick={onClick}
      >
        <div className="c-explorer__header__title__inner">
          <Icon
            name={isRoot ? 'home' : 'arrow-left'}
            className="icon--explorer-header"
          />
          <span>{page.admin_display_title || gettext('Pages')}</span>
        </div>
      </Button>
      {!isSiteRoot &&
        page.meta.locale &&
        page.translations &&
        page.translations.size > 0 && (
          <SelectLocale
            locale={page.meta.locale}
            translations={page.translations}
            gotoPage={gotoPage}
          />
        )}
    </div>
  );
};

export default ExplorerHeader;
