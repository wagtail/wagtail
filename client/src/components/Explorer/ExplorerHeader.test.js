import React from 'react';
import { mount, shallow } from 'enzyme';

import ExplorerHeader from './ExplorerHeader';

const mockProps = {
  page: {},
  depth: 2,
  transitionName: 'pop',
  onClick: jest.fn(),
};

describe('ExplorerHeader', () => {
  it('exists', () => {
    expect(ExplorerHeader).toBeDefined();
  });

  it('basic', () => {
    expect(shallow(<ExplorerHeader {...mockProps} />)).toMatchSnapshot();
  });

  it('#depth at root', () => {
    expect(shallow(<ExplorerHeader {...mockProps} depth={1} />)).toMatchSnapshot();
  });

  it('#page', () => {
    const wrapper = shallow(<ExplorerHeader {...mockProps} page={{ id: 'a', admin_display_title: 'test' }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('#onClick', () => {
    const wrapper = mount(<ExplorerHeader {...mockProps} />);
    wrapper.find('Button').simulate('click');

    expect(mockProps.onClick).toHaveBeenCalledTimes(1);
  });
});
