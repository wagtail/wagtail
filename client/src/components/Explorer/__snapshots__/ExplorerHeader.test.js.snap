// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplorerHeader #depth at root 1`] = `
<Button
  accessibleLabel={null}
  className="c-explorer__header"
  href="/admin/pages/"
  icon=""
  isLoading={false}
  onClick={[Function]}
  preventDefault={true}
  target={null}
>
  <div
    className="c-explorer__header__inner"
  >
    <Icon
      className=""
      name="home"
      title={null}
    />
    <span>
      Pages
    </span>
  </div>
</Button>
`;

exports[`ExplorerHeader #page 1`] = `
<Button
  accessibleLabel={null}
  className="c-explorer__header"
  href="/admin/pages/a/"
  icon=""
  isLoading={false}
  onClick={[Function]}
  preventDefault={true}
  target={null}
>
  <div
    className="c-explorer__header__inner"
  >
    <Icon
      className=""
      name="arrow-left"
      title={null}
    />
    <span>
      test
    </span>
  </div>
</Button>
`;

exports[`ExplorerHeader basic 1`] = `
<Button
  accessibleLabel={null}
  className="c-explorer__header"
  href="/admin/pages/"
  icon=""
  isLoading={false}
  onClick={[Function]}
  preventDefault={true}
  target={null}
>
  <div
    className="c-explorer__header__inner"
  >
    <Icon
      className=""
      name="arrow-left"
      title={null}
    />
    <span>
      Pages
    </span>
  </div>
</Button>
`;
